### ISSUE TITLE
[Bug] Param 'Seconds' out of range when adding KSP instance

### ISSUE THREAD

**Issue by scul86** [KSP-CKAN/CKAN#3500]
Background
----------
* Operating System: Manjaro w/ Stable repositories and all updates installed as of posting
* CKAN Version: 1.30.4
* KSP Version: 1.12.3
* mono version: 6.12.0, installed from Manjaro repos

**Have you made any manual changes to your GameData folder (i.e., not via CKAN)?**
no

Problem
-------
Try to add game folder for 1.12.3 to CKAN, after opening game folder via Add Instance dialog box, CKAN crashes with below error.
KSP 1.12.3 folder was freshly unzipped after download from KSP/Private Division website.
KSP Version 1.11.1 works without issue

**Steps to reproduce**
* run ckan in terminal via `mono ckan.exe`
* File -> Manage Game Instances -> New Game Instance -> Add Instance to CKAN
* In `Open` dialog box, Select 1.12.3 game folder
* Click 'open' button in dialog box *or* double click the 1.12.3 game folder
* See error

**Expected behavior**
Folder opens to enable adding game instance to CKAN

**Screenshots (if applicable)**
Will crash after clicking `Open` or double clicking highlighted folder.
![ckan](https://user-images.githubusercontent.com/11411544/147528872-c98bdcb8-85e3-4f56-993f-b579578fa5f3.png)

**CKAN error code (if applicable):**

```
9920 [6] ERROR CKAN.ErrorDialog (null) - Unhandled exception:
System.ArgumentOutOfRangeException: Valid values are between -62135596800 and 253402300799, inclusive.
Parameter name: seconds
  at System.DateTimeOffset.FromUnixTimeSeconds (System.Int64 seconds) [0x00043] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.IO.FileStatus.UnixTimeToDateTimeOffset (System.Int64 seconds, System.Int64 nanoseconds) [0x00000] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.IO.FileStatus.GetLastAccessTime (System.ReadOnlySpan`1[T] path, System.Boolean continueOnError) [0x00018] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.IO.FileSystemInfo.get_LastAccessTimeCore () [0x00011] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.IO.FileSystemInfo.get_LastAccessTimeUtc () [0x00000] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.IO.FileSystemInfo.get_LastAccessTime () [0x00000] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileSystemInfo.get_LastAccessTime()
  at System.Windows.Forms.FileSystem.GetDirectoryFSEntry (System.IO.DirectoryInfo dirinfo, System.Windows.Forms.FSEntry topFolderFSEntry) [0x00047] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Windows.Forms.FileSystem.GetNormalFolderContent (System.String from_folder, System.Collections.Specialized.StringCollection filters, System.Collections.ArrayList& directories_out, System.Collections.ArrayList& files_out) [0x00023] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Windows.Forms.FileSystem.GetFolderContent (System.Collections.Specialized.StringCollection filters, System.Collections.ArrayList& directories_out, System.Collections.ArrayList& files_out) [0x00155] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Windows.Forms.MWFVFS+WorkerThread.GetFolderContentThread () [0x00000] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00014] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ThreadHelper.ThreadStart () [0x00008] in <62b430b945fa49a19a75382ef03e7bed>:0  
[ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentOutOfRangeException: Valid values are between -62135596800 and 253402300799, inclusive.
Parameter name: seconds
  at System.DateTimeOffset.FromUnixTimeSeconds (System.Int64 seconds) [0x00043] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.IO.FileStatus.UnixTimeToDateTimeOffset (System.Int64 seconds, System.Int64 nanoseconds) [0x00000] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.IO.FileStatus.GetLastAccessTime (System.ReadOnlySpan`1[T] path, System.Boolean continueOnError) [0x00018] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.IO.FileSystemInfo.get_LastAccessTimeCore () [0x00011] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.IO.FileSystemInfo.get_LastAccessTimeUtc () [0x00000] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.IO.FileSystemInfo.get_LastAccessTime () [0x00000] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileSystemInfo.get_LastAccessTime()
  at System.Windows.Forms.FileSystem.GetDirectoryFSEntry (System.IO.DirectoryInfo dirinfo, System.Windows.Forms.FSEntry topFolderFSEntry) [0x00047] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Windows.Forms.FileSystem.GetNormalFolderContent (System.String from_folder, System.Collections.Specialized.StringCollection filters, System.Collections.ArrayList& directories_out, System.Collections.ArrayList& files_out) [0x00023] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Windows.Forms.FileSystem.GetFolderContent (System.Collections.Specialized.StringCollection filters, System.Collections.ArrayList& directories_out, System.Collections.ArrayList& files_out) [0x00155] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Windows.Forms.MWFVFS+WorkerThread.GetFolderContentThread () [0x00000] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00014] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ThreadHelper.ThreadStart () [0x00008] in <62b430b945fa49a19a75382ef03e7bed>:0
```

**Comment by HebaruSan** [comment]
Duplicate of #3473

**Comment by HebaruSan** [comment]
See https://github.com/KSP-CKAN/CKAN/issues/3473#issuecomment-962711841 for the last time this was reported, apparently the access times of the KSP ZIP can be weird (on Arch only, for an as yet unknown reason).

**Comment by scul86** [comment]
Thanks, that worked.  Sorry for the dupe.

From that thread, command that fixed it: `touch KSP_1.12.3/*`

**Comment by HebaruSan** [comment]
Cool! Maybe we should add that to [the wiki](https://github.com/KSP-CKAN/CKAN/issues/3473#issuecomment-962711841)...

**Comment by scul86** [comment]
I would edit the wiki, but seems to be locked to me... Here's a template if you want to add it:

https://github.com/KSP-CKAN/CKAN/wiki/Installing-CKAN-on-Arch

# Troubleshooting
## Crash when opening directory involving FromUnixTimeSeconds
Access times on the 1.12.x directory are mangled after extraction from the .zip downloaded from KSP/Private Division, leading to the following error - `System.ArgumentOutOfRangeException: Valid values are between -62135596800 and 253402300799, inclusive.
Parameter name: seconds`.  

`touch` the extracted folder and re-run CKAN.
```
touch KSP_linux/*
mono ckan.exe
```

**Comment by scul86** [comment]
Thinking further, I wonder if it's due to having `noatime` mount option for my home folder...

@waterlubber, do you have `noatime` mount option as well?

**Comment by HebaruSan** [comment]
> I would edit the wiki, but seems to be locked to me...

Yeah, we locked it to team members only after some repeated vandalism. Unfortunately GitHub currently doesn't have a safe way to re-open it; we just have to guess when the perpetrator has lost interest and take our chances.

> Here's a template if you want to add it:

Thanks! Added something more or less equivalent.
